test/e2e/serial/workload_placement_test.go

```diff
@@ -0,0 +1,153 @@
+
+var _ = Describe("[serial][disruptive][scheduler] workload placement", func() {
+	var testFixture *e2efixture.Fixture
+	var nroSchedObj *nropv1alpha1.NUMAResourcesScheduler
+	var schedulerName string
+	var nodeResourceTopologyList nrtv1alpha1.NodeResourceTopologyList
+	var nodeResourceTopologies []nrtv1alpha1.NodeResourceTopology
+
+	BeforeEach(func() {
+		var err error
+
+		By("Setup namespaces")
+		testFixture, err = e2efixture.Setup("e2e-test-workload-placement")
+		Expect(err).NotTo(HaveOccurred(), "unable to setup test fixture")
+
+		By("Check for NumaResourceScheduler")
+		nroSchedObj = &nropv1alpha1.NUMAResourcesScheduler{}
+		nroSchedObjKey := client.ObjectKey{Name: nrosched.NROSchedObjectName}
+		err = testFixture.Client.Get(context.TODO(), nroSchedObjKey, nroSchedObj)
+		Expect(err).NotTo(HaveOccurred(), "cannot get %q scheduler from the cluster", nrosched.NROSchedObjectName)
+
+		schedulerName = nroSchedObj.Status.SchedulerName
+		Expect(schedulerName).NotTo(BeEmpty(), "cannot autodetect TAS name from the NRO Scheduler object")
+
+		By("Check worker nodes topology is ok to run the tests")
+		err = testFixture.Client.List(context.TODO(), &nodeResourceTopologyList)
+		Expect(err).NotTo(HaveOccurred(), "unable to find NodeResourceTopolyList in the cluster")
+
+		topoloyManagerPolicy := nrtv1alpha1.SingleNUMANodeContainerLevel
+		nodeResourceTopologies = e2enrt.FilterTopologyManagerPolicy(nodeResourceTopologyList.Items, topoloyManagerPolicy)
+
+		if len(nodeResourceTopologies) == 0 {
+			Skip(fmt.Sprintf("cannot find at least one node with policy %q", string(topoloyManagerPolicy)))
+		}
+	})
+
+	AfterEach(func() {
+		err := e2efixture.Teardown(testFixture)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	Context("Cluster with multiple worker nodes suitable", func() {
+
+		BeforeEach(func() {
+
+			By("Getting nodes from NodeGroups")
+			nrOperatorObj := &nropv1alpha1.NUMAResourcesOperator{}
+			nrOperatorObjKey := client.ObjectKey{Name: NROperatorObjectName}
+			err := testFixture.Client.Get(context.TODO(), nrOperatorObjKey, nrOperatorObj)
+			Expect(err).NotTo(HaveOccurred(), "cannot get %q operator from the cluster", NROperatorObjectName)
+
+			nodes, err := getNodesFromNodeGroups(context.TODO(), testFixture.Client, nrOperatorObj.Spec.NodeGroups)
+			Expect(err).NotTo(HaveOccurred(), "cannot get Node information from NodeGroups in  %q operator", NROperatorObjectName)
+
+			By("Checking: There should be two or more nodes")
+			if len(nodes) < 2 {
+				Skip("There should be at least 2 nodes to run this test")
+			}
+
+			By("Checking: at least two nodes have at least two NUMA Zones")
+			suitableNodes, err := filterNodesWithMoreThanOneNUMAZone(nodes, nodeResourceTopologies)
+			Expect(err).NotTo(HaveOccurred())
+
+			if len(suitableNodes) < 2 {
+				Skip("At least one node should have more than one NUMA Zone to run this test")
+			}
+
+			//TODO : filter nodes with SingleNumaNode policy?
```

Comment:
you pretty much have this for free if
1. filter NRTs having the desired policy. If < 2, bail out (skip/error)
2. filter NRTs with >= 2 numa zones. If < 2, bail out (skip/error)
3. the set you have now fullfills all the requirements. Additionally, we can rely on the fact we can unambiguosly (and easily) the node objects from the NRT objects, if needed.

for our purposes I think having only the node names is sufficient.

Reply:
Changed